src/main/java/kitchenpos/table/domain/TableGroup.java

```diff
@@ -0,0 +1,105 @@
+package kitchenpos.table.domain;
+
+import kitchenpos.table.exception.AlreadyExistTableGroupException;
+import kitchenpos.table.exception.InvalidMinOrderTableSizeException;
+import org.springframework.data.annotation.CreatedDate;
+
+import javax.persistence.Column;
+import javax.persistence.Embedded;
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.Table;
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Objects;
+
+@Table(name = "table_group")
+@Entity
+public class TableGroup {
+    private static final int MIN_ORDER_TABLE_SIZE = 2;
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @CreatedDate
+    @Column(name = "created_date")
+    private LocalDateTime createdDate;
+
+    @Embedded
+    private OrderTables orderTables = new OrderTables();
+
+    protected TableGroup() {
+    }
+
+    public TableGroup(final List<OrderTable> orderTables) {
+        this(null, orderTables);
+    }
+
+    public TableGroup(final Long id, final List<OrderTable> orderTables) {
+        validateOrderTablesSize(orderTables);
+        this.id = id;
+        orderTables.forEach(this::addOrderTable);
+    }
+
+    private void validateOrderTablesSize(final List<OrderTable> orderTables) {
+        if (orderTables.size() < MIN_ORDER_TABLE_SIZE) {
+            throw new InvalidMinOrderTableSizeException(String.format("단체 지정시 테이블은 최소 %d개 이상이어야 합니다.", MIN_ORDER_TABLE_SIZE));
+        }
+    }
+
+    private void addOrderTable(final OrderTable orderTable) {
+        validateOrderTable(orderTable);
+        this.orderTables.add(orderTable);
+        orderTable.addedBy(this);
+    }
+
+    private void validateOrderTable(final OrderTable orderTable) {
+        if (orderTable.getTableGroup() != null) {
```

Comment:
getter를 사용하기 보다는 객체에 메세지를 보내보면 좋을 것 같아요! 😄 
```suggestion
        if (orderTable.nonEmpty()) {
```

Reply:
감사합니다! 부족한 부분이 정말 많았네요... 주문 테이블에 테이블그룹이 비어있는지 메시지를 던져서 물어보도록 수정하였습니다!